pkg/kubelet/kubelet.go

```diff
@@ -2244,6 +2244,27 @@ func (kl *Kubelet) getPodsToSync() []*v1.Pod {
 	return podsToSync
 }
 
+// Ensure node is registered before syncing static pods.
+func (kl *Kubelet) syncStaticPodsOnNodeRegistration() {
+	nodeCh := make(chan struct{})
+	wait.Until(func() {
+		if _, err := kl.GetNode(); err != nil {
+			return
+		}
+		close(nodeCh)
+	}, time.Second, nodeCh)
+
+	allPods := kl.podManager.GetPods()
+	staticPods := []*v1.Pod{}
+	for _, pod := range allPods {
+		if kubetypes.IsStaticPod(pod) {
+			staticPods = append(staticPods, pod)
+		}
+	}
+
+	kl.HandlePodSyncs(staticPods)
```

Comment:
hmm, I don't think here we necessarily need to wait for the full pod to sync (which can add additional latency), but rather only the mirror pod to be created. Is there any way we can check if the mirror pod is created (but avoid waiting for full sync?)